### PR TITLE
Basic tests for a freestyle project and workflow integration

### PR DESCRIPTION
**Pull request by vaskrist**
Hi, I saw that you are working on upgrading logparser and wanted to help. Would this be how a JUnit test looks like for jenkins plugins?

### CODE REVIEW

**Comment by recena** [comment]
@vaskrist what a coincidence! I was working on this plugin today. Thanks for your help.

**Comment by recena** [comment]
@vaskrist Could you review your PR?

File: src/main/java/hudson/plugins/logparser/LogParserPublisher.java [hunk 1]
```diff
@@ -231,11 +232,64 @@ public BuildStepMonitor getRequiredMonitorService() {
...
             return new LogParserProjectAction(project);
         } else {
             return null;
         }
     }
 
+	public boolean isUnstableOnWarning() {
+		return unstableOnWarning;
+	}
+	@DataBoundSetter
```
Comment (recena) [inline src/main/java/hudson/plugins/logparser/LogParserPublisher.java]: newline

File: src/test/java/hudson/plugins/logparser/BaseLogTest.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.junit.Test;
+
+import hudson.model.FreeStyleProject;
+import hudson.model.Result;
+import hudson.model.Run;
+
+public class BaseLogTest extends LogParserTestCase {
+    @Test
```
Comment (recena) [inline src/test/java/hudson/plugins/logparser/BaseLogTest.java:12]: newline

**Comment by vaskrist** [comment]
OK, I am bad at merging, I rebased my changes on your branch and fixed the issues. Is it fine like this?

**Comment by recena** [comment]
@vaskrist Thanks so much.

**Comment by recena** [comment]
@vaskrist It seems there is an error:

```
Running hudson.plugins.logparser.BaseLogTest
Running InjectedTest
Tests run: 7, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 9.239 sec - in InjectedTest
Tests run: 4, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 12.716 sec <<< FAILURE! - in hudson.plugins.logparser.BaseLogTest
testFreeStyleProjectWithLogParserStep(hudson.plugins.logparser.BaseLogTest)  Time elapsed: 1.91 sec  <<< FAILURE!
org.junit.ComparisonFailure: expected:<[1]> but was:<[2]>
    at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
    at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:57)
    at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
    at hudson.plugins.logparser.LogParserTestCase.assertLogParserCorrect(LogParserTestCase.java:59)
    at hudson.plugins.logparser.BaseLogTest.testFreeStyleProjectWithLogParserStep(BaseLogTest.java:45)


Results :

Failed tests: 
  BaseLogTest.testFreeStyleProjectWithLogParserStep:45->LogParserTestCase.assertLogParserCorrect:59 expected:<[1]> but was:<[2]>

Tests run: 11, Failures: 1, Errors: 0, Skipped: 0
```

**Comment by vaskrist** [comment]
I will look into it and fix it. I have not run the tests on Linux lately, so that might be a problem with the Linux command in freestyle project. I used it to create the echoes with INFO, WARN, ERROR and the command line itself it is probably showing up as one more error. I will fix it and improve the test messages.

**Pull request merged**